econplayground/templates/main/graph_list.html

```diff
@@ -26,40 +26,42 @@ <h2>All Graphs</h2>
                                 {% if graph.description %}
                                 <p>{{ graph.description }}</p>
                                 {% endif %}
-                                {% if i_am_instructor and graph.is_published %}<span class="badge badge-pill badge-primary">Published</span>{% endif %}
-                                {% if i_am_instructor and graph.needs_submit %}<span class="badge badge-pill badge-secondary">LTI</span>{% endif %}
+                                {% if i_am_instructor and graph.is_published %}
+                                    <span class="badge badge-pill badge-primary">Published</span>
+                                {% endif %}
+                                {% if i_am_instructor and graph.needs_submit %}
+                                    <span class="badge badge-pill badge-secondary">LTI</span>
+                                {% endif %}
                             </div>
                             <div class="card-footer"><small><a href="?topic={{ graph.topic }}">{{ graph.topic }}</a></small></div>
-                        </div> 
-                    </div> 
+                        </div>
+                    </div>
                     {% empty %}
                     No graphs here.
                     {% endfor %}
                 {% elif not active_topic %} {# Renders all graphs #}
                 <div class="col-xl-12 col-lg-12 col-md-12 col-sm-12 col-xs-12">
                     {# First render graphs not assigned to a topic #}
-                    {% for graph in graph_list%}
-                        {% if not graph.topic %}
-                            <a href="{% url 'graph_detail' graph.id %}">{{ graph.title }}</a><br/>
-                            <p>{{ graph.description }}</p><br/>
-                        {% endif %}
+                    {% for graph in graphs_without_topics %}
+                        <a href="{% url 'graph_detail' graph.id %}">{{ graph.title }}</a>
+                        <p>{{ graph.description }}</p>
                     {% endfor %}
                     {% for topic in topic_list %}
                         {# Then render those with a topic #}
                         <h3>{{ topic.name }}</h3>
-                        {% for graph in graph_list%}
-                            {% if topic.id == graph.topic.id %}
-                                <a href="{% url 'graph_detail' graph.id %}">{{ graph.title }}</a><br/>
-                                <p>{{ graph.description }}</p><br/>
+                        {% for graph in topic.graph_set.all %}
```

Comment:
@nikolas what is the `graph_set` attribute on topic? is this something that Django provides?

Reply:
yeah - foreignkeys create that way of querying items - https://docs.djangoproject.com/en/2.0/topics/db/examples/many_to_one/